src/mbgl/style/expression/array_assertion.cpp

```diff
@@ -81,6 +81,28 @@ ParseResult ArrayAssertion::parse(const Convertible& value, ParsingContext& ctx)
     ));
 }
 
+mbgl::Value ArrayAssertion::serialize() const {
+    std::vector<mbgl::Value> serialized;
+    serialized.emplace_back(getOperator());
+    getType().match(
```

Comment:
In this case, since we know it's a `type::Array`, you can do `const auto& array = getType().get<type::Array>()` instead of going through `match`